test/e2e/common/pods.go

```diff
@@ -192,8 +195,32 @@ var _ = framework.KubeDescribe("Pods", func() {
 			LabelSelector:   selector.String(),
 			ResourceVersion: pods.ListMeta.ResourceVersion,
 		}
-		w, err := podClient.Watch(options)
-		Expect(err).NotTo(HaveOccurred(), "failed to set up watch")
+
+		listCompleted := make(chan bool, 1)
+		lw := &cache.ListWatch{
+			ListFunc: func(options metav1.ListOptions) (runtime.Object, error) {
+				options.LabelSelector = selector.String()
+				podList, err := podClient.List(options)
+				if err == nil {
+					for {
```

Comment:
drop the for loop, we don't need to do this more than once

Reply:
@liggitt, removed the loop